Minio/DataModel/ObjectLegalHoldConfiguration.cs

```diff
@@ -0,0 +1,44 @@
+/*
+ * MinIO .NET Library for Amazon S3 Compatible Cloud Storage, (C) 2020 MinIO, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+using System;
+using System.Xml.Serialization;
+
+namespace Minio.DataModel
+{
+    [Serializable]
+    [XmlRoot(ElementName = "LegalHold", Namespace = "http://s3.amazonaws.com/doc/2006-03-01/")]
+
+    public class ObjectLegalHoldConfiguration
+    {
+        public ObjectLegalHoldConfiguration()
+        {
+            this.Status = null;
+        }
+
+        public ObjectLegalHoldConfiguration(bool enable = true)
+        {
+            if (enable)
+            {
+                this.Status = "ON";
```

Comment:
why can't this be a bool?

Reply:
This structure is the equivalent of the XML configuration. It needs to be serialized & deserialized. So there needs to be a one-to-one representation between the class members & XML elements.